### PR TITLE
Syntax lookup: labeled and optional args

### PR DESCRIPTION
**Pull request by fhammerschmidt**

### CODE REVIEW

**Comment by aspeddro** [comment]
Can you add the `@tag` attribute? 

https://rescript-lang.org/docs/manual/latest/variant#tagged-variants

**Comment by fhammerschmidt** [comment]
Will do, but in another PR. Added [to the list](https://github.com/rescript-association/rescript-lang.org/issues/162#issue-777444557) for now.

File: misc_docs/syntax/language_labeled_argument.mdx [hunk 1]
```diff
@@ -0,0 +1,36 @@
+---
+id: "labeled-argument"
+keywords: ["labeled", "argument"]
+name: "~arg"
+summary: "This is a `labeled argument`."
+category: "languageconstructs"
+---
+
+When declaring a function, arguments can be prefixed with `~` which means that they can and need to be called by their name, not the argument position. This is especially useful to differentiate them more easily if they are of the same type.
```
Comment (zth) [inline misc_docs/syntax/language_labeled_argument.mdx:9]: Maybe mention explicitly (and add an example) that they can now be supplied in any order? And that it still works with partial application.

File: misc_docs/syntax/language_optional_labeled_argument.mdx [hunk 2]
```diff
@@ -0,0 +1,48 @@
+---
+id: "optional-labeled-argument"
+keywords: ["optional", "labeled", "argument"]
+name: "~arg=?"
+summary: "This is an `optional labeled argument`."
+category: "languageconstructs"
+---
+
+Labeled arguments, i.e. arguments that are prefixed with `~`, can be suffixed with `=?` to denote that they are optional. Thus, they can be
```
Comment (zth) [inline misc_docs/syntax/language_optional_labeled_argument.mdx:9]: Good to mention default values here as well.

**Review by zth — COMMENTED** [review]
Nice! 2 small comments from me, not sure if they're relevant in the syntax lookup though.

**Review by zth — APPROVED** [review]

**Pull request merged**